src/HotChocolate/Core/test/Validation.Tests/FragmentSpreadIsPossibleRuleTests.cs

```diff
@@ -158,5 +125,360 @@ public void Star_Wars_With_Inline_Fragments()
+                fragment dogFragment on Dog { barkVolume }
+            ");
+        }
+
+        [Fact]
+        public void ObjectIntoContainingUnion()
+        {
+            // arrange
+            ExpectValid(@"
+                {
+                    catOrDog {
+                        ...objectWithinUnion
+                    }
+                }
+                fragment objectWithinUnion on CatOrDog { ...dogFragment }
+                fragment dogFragment on Dog { barkVolume }
+            ");
+        }
+
+        [Fact]
+        public void UnionIntoContainedObject()
+        {
+            // arrange
+            ExpectValid(@"
+                {
+                    dog {
+                        ...unionWithinObject
+                    }
+                }
+                fragment unionWithinObject on Dog { ...catOrDogFragment }
+                fragment catOrDogFragment on CatOrDog { __typename }
+            ");
+        }
+
+        [Fact]
+        public void UnionIntoOverlappingInterface()
+        {
+            // arrange
+            ExpectValid(@"
+                {
+                    human{
+                        pets {
+                        ...unionWithinInterface
+                        }
+                    }
+                }
+                fragment unionWithinInterface on Pet { ...catOrDogFragment }
+                fragment catOrDogFragment on CatOrDog { __typename }
+            ");
+        }
+
+        [Fact]
+        public void UnionIntoOverlappingUnion()
+        {
+            // arrange
+            ExpectValid(@"
+                {
+                    dogOrHuman { 
+                        ...unionWithinUnion 
+                    }
+                }
+                fragment unionWithinUnion on DogOrHuman { ...catOrDogFragment }
+                fragment catOrDogFragment on CatOrDog { __typename }
+            ");
+        }
+
+        [Fact]
+        public void InterfaceIntoImplementedObject()
+        {
+            // arrange
+            ExpectValid(@"
+                {
+                    dog {
+                        ...interfaceWithinObject
+                    }
+                }
+                fragment interfaceWithinObject on Dog { ...petFragment }
+                fragment petFragment on Pet { name }
+            ");
+        }
+
+        [Fact]
+        public void InterfaceIntoOverlappingInterface()
+        {
+            // arrange
+            ExpectValid(@"
+                {
+                    human{
+                        pets {
+                        ...interfaceWithinInterface
+                        }
+                    }
+                }
+                fragment interfaceWithinInterface on Pet { ...beingFragment }
+                fragment beingFragment on Being { name }
+            ");
+        }
+
+        [Fact]
+        public void InterfaceIntoOverlappingInterfaceInInlineFragment()
```

Comment:
@michaelstaib  is this an interface implement interface on?

Reply:
yes